controllers/usersController.ts

```diff
@@ -1,73 +1,139 @@
-import * as validate from "./utils/validate.js";
 import bcrypt from "bcrypt";
-import { PrismaClient } from "@prisma/client";
+import { PrismaClient, Prisma } from "@prisma/client";
 
-import { RequestHandler } from "express";
+import { RequestHandler, Request, Response, NextFunction } from "express";
+
+import { createAccessToken } from "./utils/auth.js";
 
 const prisma = new PrismaClient();
 
 export const getAllUsers: RequestHandler = (req, res) => {
     res.send("hit get users route");
 };
 
-export const registerUser: RequestHandler = async (req, res) => {
+export const registerUser: RequestHandler = async (
+    req: Request,
+    res: Response
+) => {
     const { email, username, firstName, lastName, password } = req.body;
 
-    const newUser = await prisma.user
-        .create({
-            data: {
-                email,
-                username,
-                first_name: firstName,
-                last_name: lastName,
-                password: bcrypt.hashSync(password, 10),
-            },
-        })
-        .then((createdUser: any) => {
-            return createdUser;
-        });
+    const newUser = await prisma.user.create({
+        data: {
+            email,
+            username,
+            first_name: firstName,
+            last_name: lastName,
+            password: bcrypt.hashSync(password, 10),
+        },
+    });
 
     if (newUser) {
         return res.json({
             heading: "Registration successful",
             message:
                 "You can now log in using your registered email or username.",
         });
-    } else {
-        return res.status(500).send({
-            heading: "Server error",
-            message: "Please try again.",
-        });
     }
+
+    return res.status(500).send({
+        heading: "Server error",
+        message: "Please try again.",
+    });
 };
 
-export const checkDuplicateEmail: RequestHandler = async (req, res, next) => {
+export const checkDuplicateEmail: RequestHandler = async (
+    req: Request,
+    res: Response,
+    next: NextFunction
+) => {
     const { email, username } = req.body;
 
-    const resultEmail = await prisma.user.findUnique({
+    const resultEmail: { email: string } | null = await prisma.user.findUnique({
         where: { email: email },
         select: {
             email: true,
         },
     });
 
-    const resultUsername = await prisma.user.findUnique({
-        where: { username: username },
-        select: {
-            username: true,
-        },
-    });
+    const resultUsername: { username: string } | null =
+        await prisma.user.findUnique({
+            where: { username: username },
+            select: {
+                username: true,
+            },
+        });
 
-    if (resultEmail || resultUsername) {
+    if (resultEmail !== null) {
```

Comment:
I think you can make it like this, if possible.
```suggestion
    if (resultEmail) {
```
since null is already one of the falsy values.

Reply:
apply this to other same scenarios as well if possible.

Reply:
Got it will apply this. I just wanted to be more strict just in case another falsy value is there but I guess the types handle that.